instance/models/openedx_instance.py

```diff
@@ -146,6 +147,21 @@ def studio_domain(self):
             return self.internal_studio_domain
 
     @property
+    def studio_domain_nginx_regex(self):
+        """
+        Regex that matches either the internal or the external Studio URL.
+
+        This is exclusively meant for the Ansible variable CMS_HOSTNAME to configure the nginx
+        server_name regex to match the Studio domains.
+        """
+        if self.external_studio_domain:
+            domains = [self.external_studio_domain, self.internal_studio_domain]
+        else:
+            domains = [self.internal_studio_domain]
+        choices = '|'.join(map(re.escape, domains))  # pylint: disable=bad-builtin
```

Comment:
@antoviaque You can always use a list comprehension (or in this case, a generator expression) instead of using `map()`.  However, I personally think `map()` is more readable in this case.  It's less verbose and doesn't introduce a loop variable, so I find it easier to grasp.

I'd be in favour of removing `map` from the `bad-functions` list in the pylint configuration.  It simply isn't bad at all; `map()` has been a core part of Python since at least version 1.5, and I don't see why we should stop using it just because someone decided to add this warning to pylint.

Pylint in default configuration complains about a lot of stuff it actually shouldn't.  We should adjust the configuration to our likings rather than taking arbitrary (and sometimes outright bad) defaults as the law.